sdk/go/common/workspace/loaders.go

```diff
@@ -121,33 +141,105 @@ type projectStackLoader struct {
 	internal map[string]*ProjectStack
 }
 
+// Rewrite config values to make them namespaced. Using the project name as the default namespace
+// for example:
+//
+//	config:
+//	  instanceSize: t3.micro
+//
+// is valid configuration and will be rewritten in the form
+//
+//	config:
+//	  {projectName}:instanceSize:t3.micro
+func stackConfigNamespacedWithProject(project *Project, projectStack map[string]interface{}) map[string]interface{} {
+	if project == nil {
+		// return the original config if we don't have a project
+		return projectStack
+	}
+
+	config, ok := projectStack["config"]
+	if ok {
+		configAsMap, isMap := config.(map[string]interface{})
+		if isMap {
+			modifiedConfig := make(map[string]interface{})
+			for key, value := range configAsMap {
+				if strings.Contains(key, ":") {
+					// key is already namespaced
+					// use it as is
+					modifiedConfig[key] = value
+				} else {
+					namespacedKey := fmt.Sprintf("%s:%s", project.Name, key)
+					modifiedConfig[namespacedKey] = value
+				}
+			}
+
+			projectStack["config"] = modifiedConfig
+			return projectStack
+		}
+	}
+
+	return projectStack
+}
+
 // Load a ProjectStack config file from the specified path. The configuration will be cached for subsequent loads.
-func (singleton *projectStackLoader) load(path string) (*ProjectStack, error) {
+func (singleton *projectStackLoader) load(project *Project, path string) (*ProjectStack, error) {
 	singleton.Lock()
 	defer singleton.Unlock()
 
 	if v, ok := singleton.internal[path]; ok {
 		return v, nil
 	}
 
-	marshaler, err := marshallerForPath(path)
+	marshaller, err := marshallerForPath(path)
 	if err != nil {
 		return nil, err
 	}
 
-	var projectStack ProjectStack
 	b, err := readFileStripUTF8BOM(path)
 	if os.IsNotExist(err) {
-		projectStack = ProjectStack{
+		defaultProjectStack := ProjectStack{
 			Config: make(config.Map),
 		}
-		singleton.internal[path] = &projectStack
-		return &projectStack, nil
+		singleton.internal[path] = &defaultProjectStack
+		return &defaultProjectStack, nil
 	} else if err != nil {
 		return nil, err
 	}
 
-	err = marshaler.Unmarshal(b, &projectStack)
+	var projectStackRaw interface{}
+	err = marshaller.Unmarshal(b, &projectStackRaw)
+	if err != nil {
+		return nil, err
+	}
+
+	if projectStackRaw == nil {
```

Comment:
I am _very_ suspect of this. I assume this was to fix the "destory works without a Pulumi.yaml" but fixing it at this level means that any command using projects will try to use an empty project rather than failing validation, that feels wrong.